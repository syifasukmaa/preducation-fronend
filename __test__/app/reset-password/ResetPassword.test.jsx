import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import ResetPasswordPage from '@/app/reset-password/page'

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(() => ({ get: jest.fn(() => 'mocked_token') })),
}))

jest.mock('@/utils/fetch', () => ({
  resetPassword: jest.fn(() => Promise.resolve({ ok: true, json: () => ({}) })),
}))

describe('Reset Password Page', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('Should render', () => {
    const page = render(<ResetPasswordPage />)
    expect(page).toMatchSnapshot()
  })

  it('Should contain Text Reset Password', () => {
    render(<ResetPasswordPage />)
    expect(screen.getByRole('heading')).toHaveTextContent('Reset Password')
  })

  it('Should toggles password visibility when eye icon is clicked', () => {
    render(<ResetPasswordPage />)
    const eyeIcon = screen.getByTitle('button-visibility')
    const passwordInput = screen.getByPlaceholderText('Password')

    expect(passwordInput.type).toBe('password')

    fireEvent.click(eyeIcon)
    expect(passwordInput.type).toBe('text')
  })

  //   it('Handles form submission successfully', async () => {
  //     const resetPassword = require('@/utils/fetch').resetPassword
  //     resetPassword.mockResolvedValueOnce({ ok: true })
  //     render(<ResetPasswordPage />)

  //     console.log(resetPassword)

  //     fireEvent.change(screen.getByPlaceholderText('Password'), { target: { value: 'new_password_example' } })
  //     fireEvent.change(screen.getByPlaceholderText('Konfirmasi password'), {
  //       target: { value: 'confirm_password_example' },
  //     })
  //     fireEvent.click(screen.getByText('Ganti'))

  //     await waitFor(() => {
  //       expect(resetPassword).toHaveBeenCalledWith('mocked_token', 'new_password_example', 'confirm_password_example')
  //       expect(screen.getByText('Anda berhasil mengubah password')).toBeInTheDocument()
  //     })
  //   })
})
