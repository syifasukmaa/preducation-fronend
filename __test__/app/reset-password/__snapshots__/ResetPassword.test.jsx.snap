// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Page Should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col lg:flex-row w-full min-h-screen"
      >
        <div
          class="p-8 lg:p-16 lg:w-2/3 flex items-center justify-center"
        >
          <div
            class="w-full lg:w-2/3"
          >
            <h1
              class="font-bold text-2xl text-primary-dark-blue  mb-8 lg:mb-12 text-left"
            >
              Reset Password
            </h1>
            <div
              class="mt-2 block lg:mb-4"
            >
              <br />
              <label
                for="passInput"
              >
                Masukkan Password Baru
              </label>
              <br />
              <div
                class="relative"
              >
                <input
                  class="  border-2 rounded-lg w-full p-2 outline-none"
                  id="passInput"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
                <button
                  class="absolute top-1/2 transform -translate-y-1/2 right-4"
                  title="button-visibility"
                >
                  <svg
                    color="grey"
                    fill="currentColor"
                    height="30"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: grey;"
                    viewBox="0 0 256 256"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="block lg:mb-4"
            >
              <label
                for="confirmPassInput"
              >
                Ulangi Password Baru
              </label>
              <br />
              <div
                class="relative"
              >
                <input
                  class=" border-2 rounded-lg w-full p-2 outline-none"
                  id="confirmPassInput"
                  name="confirmPassword"
                  placeholder="Konfirmasi password"
                  required=""
                  type="password"
                  value=""
                />
                <button
                  class="absolute top-1/2 transform -translate-y-1/2 right-4"
                >
                  <svg
                    color="grey"
                    fill="currentColor"
                    height="30"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: grey;"
                    viewBox="0 0 256 256"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <br />
            <button
              class="cursor-not-allowed text-white bg-orange-05 rounded-xl w-full h-10  mb-10 flex items-center justify-center"
              disabled=""
            >
              Ganti
            </button>
            <br />
          </div>
        </div>
        <div
          class="bg-primary-dark-blue p-8 lg:p-16 lg:w-1/3 flex items-center justify-center"
        >
          <img
            alt="logo"
            class="mx-auto"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="150"
            src="/_next/image?url=%2Fimg%2FiconPreducation.png&w=384&q=75"
            srcset="/_next/image?url=%2Fimg%2FiconPreducation.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2FiconPreducation.png&w=384&q=75 2x"
            style="color: transparent;"
            width="150"
          />
        </div>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col lg:flex-row w-full min-h-screen"
    >
      <div
        class="p-8 lg:p-16 lg:w-2/3 flex items-center justify-center"
      >
        <div
          class="w-full lg:w-2/3"
        >
          <h1
            class="font-bold text-2xl text-primary-dark-blue  mb-8 lg:mb-12 text-left"
          >
            Reset Password
          </h1>
          <div
            class="mt-2 block lg:mb-4"
          >
            <br />
            <label
              for="passInput"
            >
              Masukkan Password Baru
            </label>
            <br />
            <div
              class="relative"
            >
              <input
                class="  border-2 rounded-lg w-full p-2 outline-none"
                id="passInput"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <button
                class="absolute top-1/2 transform -translate-y-1/2 right-4"
                title="button-visibility"
              >
                <svg
                  color="grey"
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: grey;"
                  viewBox="0 0 256 256"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="block lg:mb-4"
          >
            <label
              for="confirmPassInput"
            >
              Ulangi Password Baru
            </label>
            <br />
            <div
              class="relative"
            >
              <input
                class=" border-2 rounded-lg w-full p-2 outline-none"
                id="confirmPassInput"
                name="confirmPassword"
                placeholder="Konfirmasi password"
                required=""
                type="password"
                value=""
              />
              <button
                class="absolute top-1/2 transform -translate-y-1/2 right-4"
              >
                <svg
                  color="grey"
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: grey;"
                  viewBox="0 0 256 256"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <br />
          <button
            class="cursor-not-allowed text-white bg-orange-05 rounded-xl w-full h-10  mb-10 flex items-center justify-center"
            disabled=""
          >
            Ganti
          </button>
          <br />
        </div>
      </div>
      <div
        class="bg-primary-dark-blue p-8 lg:p-16 lg:w-1/3 flex items-center justify-center"
      >
        <img
          alt="logo"
          class="mx-auto"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="150"
          src="/_next/image?url=%2Fimg%2FiconPreducation.png&w=384&q=75"
          srcset="/_next/image?url=%2Fimg%2FiconPreducation.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2FiconPreducation.png&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="Toastify"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
